blockprod/src/detail/mod.rs

```diff
@@ -474,4 +480,4 @@ fn generate_finalize_block_data(
 }
 
 #[cfg(test)]
-mod tests;
+pub mod tests;
```

Comment:
This should never be done. Exposing test modules guarded by cfg(test) can lead to all kinds of problems, including the false perception that something is public, while it's not.